Mathlib/MeasureTheory/Measure/EverywherePos.lean

```diff
@@ -0,0 +1,296 @@
+/-
+Copyright (c) 2024 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.MeasureTheory.Group.Measure
+import Mathlib.Topology.UrysohnsLemma
+
+/-!
+# Everywhere positive sets in measure spaces
+
+A set `s` in a topological space with a measure `μ` is *everywhere positive* (also called
+*self-supporting*) if any neighborhood `n` of any point of `s` satisfies `μ (s ∩ n) > 0`.
+
+## Main definitions and results
+
+* `μ.IsEverywherePos s` registers that, for any point in `s`, all its neighborhoods have positive
+  measure inside `s`.
+* `μ.everywherePosSubset s` is the subset of `s` made of those points all of whose neighborhoods
+  have positive measure inside `s`.
+* `everywherePosSubset_ae_eq` shows that `s` and `μ.everywherePosSubset s` coincide almost
+  everywhere if `μ` is inner regular and `s` is measurable.
+* `isEverywherePos_everywherePosSubset` shows that `μ.everywherePosSubset s` satisfies the property
+  `μ.IsEverywherePos` if `μ` is inner regular and `s` is measurable.
+
+The latter two statements have also versions when `μ` is inner regular for finite measure sets,
+assuming additionally that `s` has finite measure.
+
+* `IsEverywherePos.IsGδ` proves that an everywhere positive compact closed set is a Gδ set,
+  in a topological group with a left-invariant measure. This is a nontrivial statement, used
+  crucially in the study of the uniqueness of Haar measures.
+* `innerRegularWRT_preimage_one_hasCompactSupport_measure_ne_top`: for a Haar measure, any
+  finite measure set can be approximated from inside by level sets of continuous
+  compactly supported functions. This property is also known as completion-regularity of Haar
+  measures.
+-/
+
+open scoped Topology ENNReal NNReal
+open Set Filter
+
+namespace MeasureTheory.Measure
+
+variable {α : Type*} [TopologicalSpace α] [MeasurableSpace α]
+
+/-- A set `s` is *everywhere positive* (also called *self-supporting*) with respect to a
```

Comment:
Isn't there a notion of support for a measure? This seems to be equivalent to the fact that the support of the restriction of the measure to s is equal to s.

Reply:
(By reading the file further down, I understand that you're defining that, without defining the support of the measure in general)

Reply:
Yes, I could have phrased it in terms of support, by requiring that the support of the restriction of the measure to `s` contains `s`.